### ISSUE TITLE
Model progress stops after 200th epoch (using OC-MAML run_MNIST_example.sh)

### ISSUE THREAD

**Issue by chriskok** [AhmedFrikha/Few-Shot-One-Class-Classification-via-Meta-Learning#1]
Hi! I'm running into an issue where the model progress stops/is in an infinite loop/is very slow after the 200th epoch with the MNIST example. Here's what I've done so far:

- Downloaded data from: http://yann.lecun.com/exdb/mnist/
- Updated base data filepaths for all relevant files
- Ran the Few-Shot-One-Class-Classification-via-Meta-Learning/MAMLs_Reptiles/MNIST/generate_tasks python files to generate all 9 MNIST validation tests sets 
- Ran Few-Shot-One-Class-Classification-via-Meta-Learning/MAMLs_Reptiles/MNIST/metalearning_algorithms/run_MNIST_example.sh

Epoch 0 to 100 and 100 to 200 were relatively quick (within minutes) but I waited for a couple hours and the model hasn't made progress beyond that point. Was wondering if there was an issue or if I should just be more patient. Either way, I really appreciate you making the code public and I'm glad you all were accepted to AAAI 2021! 

Here's how it looks when halted:
![MAML_loop](https://user-images.githubusercontent.com/25407259/107063554-2221c980-67a0-11eb-8b40-c40f958cbf57.png)

Keyboard interrupting shows that it's always stuck at the maml_class.py metatrain_op function specifically when sess.run is called:
![MAML_stopped](https://user-images.githubusercontent.com/25407259/107063561-23eb8d00-67a0-11eb-8f6d-97b3704e1765.png)

Please let me know if there's a better way to check what the problem is as well. Thanks again!

**Comment by AhmedFrikha** [comment]
Hi, thanks for your interest in our paper. The prints you see are only made, if a lower validation loss than the minimum (in your case 0.23) is computed and the validation episodes are conducted every 100 iterations. So the meta-training is still running but no lower validation loss was computed. 

OC-MAML converges quite quickly on multi-MNIST. So my advice would be to reduce the meta-epochs in file config_MNIST_ocmaml... .json from 10000 to 1000 or 2000. You can also increase the chances of finding a lower validation loss by conducting meta-validation episodes more often. For that reduce the "val_interval" (line 358 in main.py).

**Comment by chriskok** [comment]
AH okay understood. That makes sense now, I'm still new to meta-learning so this really helps. Appreciate you getting back to me so quickly! 

Closing this thread as it is no longer an issue.
